### PR TITLE
lint: require "await" or .catch() on async calls

### PR DESCRIPTION
**Pull request by justinmk3**
Continued from:

- #1638
- #4202
- #4218
- #4220

Solution:
- Enable eslint rule: https://typescript-eslint.io/rules/no-floating-promises/

## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -93,14 +93,13 @@ module.exports = {
         'no-null/no-null': 'error',
         '@typescript-eslint/no-empty-function': 'off',
         '@typescript-eslint/no-unused-vars': 'off',
-        // New rules --> New TODOs
+        '@typescript-eslint/no-floating-promises': 'error', // Promises must catch errors or be awaited.
```
Comment (damntrecky) [inline .eslintrc.js:96]: Question: Should this rule be applied to only non test files? Maybe excluding files with the word `test` or `Test` in them. I don't think there is the `.spec.` pattern to apply to in this codebase.

Comment (justinmk3) [reply .eslintrc.js:96]: That would add a source of accidents to the tests. What would be gained?

File: src/lambda/config/templates.ts [hunk 2]
```diff
@@ -197,7 +197,7 @@ export function showTemplatesConfigurationError(
 }
 
 export async function ensureTemplatesConfigFileExists(path: string): Promise<void> {
-    ensureDir(_path.dirname(path))
+    await ensureDir(_path.dirname(path))
```
Comment (justinmk3) [inline src/lambda/config/templates.ts:200]: This looks like a potential bug (race).

File: .eslintrc.js (hunk 1 above)
Comment (justinmk3) [reply .eslintrc.js]: @damntrecky here's a recent example where tests were not correctly await'd: https://github.com/aws/aws-toolkit-vscode/pull/4240/files#r1444539552

**Pull request merged**